### PR TITLE
Refactor/stop using immer

### PR DESCRIPTION
**Pull request by solidgoldpig**
https://dsdmoj.atlassian.net/browse/FB-495

### CODE REVIEW

File: templates/nunjucks/page.editable.njk.html [hunk 1]
```diff
@@ -479,7 +481,7 @@
    
       blockContent = `<div class="fb-block-instance-properties"><span class="fb-block-instance-properties-heading fb-block-edit">${blockText}</span><ul>`
       blockContent += `<li class="fb-block-instance-properties-type-item"><span class="fb-block-instance-properties-type">${blockTypeDisplay}</span></li>`
-      if (blockHasName || blockType === 'radios' || blockType === 'date') {
+      if (blockHasName || blockType === 'radios' || blockType === 'date' || blockType === 'fileupload') {
```
Comment (solidgoldpig) [inline templates/nunjucks/page.editable.njk.html]: Enable fileupload components to have direct access to name property

File: templates/nunjucks/page.editable.njk.html [hunk 2]
```diff
@@ -489,7 +491,7 @@
         const blockValueProperty = `<li><a href="/admin/instance/${blockLink}/value">Set <span class="fb-block-instance-property-link">${blockType} ${blockTypeName}</span> value</a></li>`
         blockContent += blockValueProperty
       }
-      if (blockType.match(/^(fileupload)$/)) {
+      if (blockType === 'fileupload') {
```
Comment (solidgoldpig) [inline templates/nunjucks/page.editable.njk.html]: No need for regex

File: templates/nunjucks/page.editable.njk.html [hunk 3]
```diff
@@ -320,7 +321,8 @@
   clip-path: unset !important;
   white-space: normal !important;
 }
-.govuk-visually-hidden::before {
+legend.govuk-visually-hidden::before,
```
Comment (solidgoldpig) [inline templates/nunjucks/page.editable.njk.html]: Only need this info on questions that have been visually hidden

File: lib/admin/route/admin.instance.create.id/admin.instance.create.id.controller.js [hunk 4]
```diff
@@ -148,6 +143,12 @@ instanceCreateIdController.validate = (pageInstance, pageData, POST) => {
       const schemaProperty = schema.properties[idSeed]
       let validationError = validator.validate(idSeedValue, schemaProperty).errors[0]
       if (validationError) {
+        validationError.values = {
```
Comment (solidgoldpig) [inline lib/admin/route/admin.instance.create.id/admin.instance.create.id.controller.js]: Ensure that error messages when creating ids for new instances get properly interpolated

File: lib/admin/admin.js [hunk 5]
```diff
@@ -229,18 +231,12 @@ const pageHandler = (req, res, next) => {
 }
 
 const renderAdminPage = (res, req, next, pageInstance, pageData) => {
-  // Update name values
+  pageInstance = setFormContent(pageInstance, pageData)
```
Comment (solidgoldpig) [inline lib/admin/admin.js]: Perform same step in admin views as for standard runner views

File: lib/admin/route/admin.instance.create.id/admin.instance.create.id.controller.js [hunk 6]
```diff
@@ -95,20 +92,17 @@ instanceCreateIdController.setData = (pageInstance, pageData) => {
...
-          })
-          const existingRadios = jp.query(draft, '$..[?(@._id == "admin.instance.create.id--existing--radios")]')[0]
-          existingRadios.items = existingItems
-          return draft
+        const existingItems = existingIds.map((_id, index) => {
+          const item = {
+            _id: `admin.instance.create.id--existing--radios--${index}`,
+            _type: 'radio',
+            value: _id,
+            label: getInstanceTitle(_id)
```
Comment (tommotaylor) [inline lib/admin/route/admin.instance.create.id/admin.instance.create.id.controller.js]: The three uses of `_id` here, is the use on L97 meant to be a string? L99 and L100 look like values. Is this not going to blow up? Or are they all values?

Comment (solidgoldpig) [reply lib/admin/route/admin.instance.create.id/admin.instance.create.id.controller.js]: L97 - setting the _id of the radio option component (every component has an _id - even though this one is being dynamically generated)
L99 - setting the value of the radio option to be the _id of the component/page we're looping over
L110 - set the label to be the title value of the component/page we're looping over

**Pull request merged**